library/connect.c

```diff
@@ -818,7 +820,7 @@ int ziti_channel_start_connection(struct ziti_conn *conn) {
             },
             {
                     .header_id = CallerIdHeader,
-                    .length = strlen(conn->ziti_ctx->session->identity->name),
+                    .length = strlen(conn->ziti_ctx->session->identity->name) + 1,
```

Comment:
this is not needed -- we shouldn't put '\0' byte into the header, it does not translate into other languages correctly